crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -178,21 +180,74 @@ impl SimpleFaucet {
         Ok(responses)
     }
 
-    async fn transfer_sui(
+    async fn construct_transfer_sui_txn_with_retry(
         &self,
         coin_id: ObjectID,
         signer: SuiAddress,
         recipient: SuiAddress,
         budget: u64,
         amount: u64,
         uuid: Uuid,
-    ) -> Result<SuiTransactionResponse, anyhow::Error> {
-        let context = &self.wallet;
+    ) -> Result<TransactionData, anyhow::Error> {
+        // try 3 times in total
+        let retry_intervals_ms = [Duration::from_millis(500), Duration::from_millis(1000)];
```

Comment:
nit: I would use constants or a counter here so its easy to see that it retries 3 times.

Reply:
with a counter it's hard to see the intervals. I'd prefer to keep it in this way and change the comment to "retry 2 times"